### PR TITLE
Converted comments to google; fixed pointer tensor serde and send/get functionality

### PR DESCRIPTION
**Pull request by iamtrask**
fixes https://github.com/OpenMined/PySyft/issues/1694

### CODE REVIEW

File: syft/serde.py [hunk 1]
```diff
@@ -496,111 +498,61 @@ def _detail_ellipsis(ellipsis: bytes) -> Ellipsis:
...
+    )
 
-def _simplify_pointer_tensor(ptr: PointerTensor) -> Dict:
-    """
-    This function takes the attributes of a PointerTensor and saves them in a dictionary
-    Args:
-        PointerTensor: a PointerTensor
-    Returns:
-        Dict: a dictionary holding the attributes of the PointerTensor
-    Usage:
```
Comment (mari-linhares) [inline syft/serde.py]: Usage apparently is not supported by sphinx/napoleon (https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html). Usage should be replaced by Example as shown in this example (https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html#example-google).

File: syft/serde.py [hunk 2]
```diff
@@ -183,6 +183,8 @@ def _detail_torch_tensor(tensor: bin) -> torch.Tensor:
         torch.Tensor: a torch tensor that was serialized
     """
 
+    id, tensor = tensor
```
[the remark this answers is not in the document]
Comment (iamtrask) [reply syft/serde.py]: Great point! Renamed to `tensor_tuple` and added better inline documentation to describe it.

File: syft/serde.py [hunk 3]
```diff
@@ -168,7 +168,7 @@ def _simplify_torch_tensor(tensor: torch.Tensor) -> bin:
     binary_stream = io.BytesIO()
     torch.save(tensor, binary_stream)
     tensor_bin = binary_stream.getvalue()
-    return tensor_bin
+    return (tensor.id, tensor_bin)
```
[the remark this answers is not in the document]
Comment (iamtrask) [reply syft/serde.py]: Done!

File: syft/serde.py [hunk 4]
```diff
@@ -496,111 +498,61 @@ def _detail_ellipsis(ellipsis: bytes) -> Ellipsis:
...
-def _simplify_pointer_tensor(ptr: PointerTensor) -> Dict:
+def _simplify_pointer_tensor(ptr: PointerTensor) -> tuple:
     """
     This function takes the attributes of a PointerTensor and saves them in a dictionary
-
     Args:
         PointerTensor: a PointerTensor
-
     Returns:
         Dict: a dictionary holding the attributes of the PointerTensor
```
[the remark this answers is not in the document]
Comment (iamtrask) [reply syft/serde.py]: Done!

File: syft/serde.py [hunk 5]
```diff
@@ -496,111 +498,61 @@ def _detail_ellipsis(ellipsis: bytes) -> Ellipsis:
...
 
-def _detail_pointer_tensor(data: Dict) -> PointerTensor:
+def _detail_pointer_tensor(data: tuple) -> PointerTensor:
     """
     This function reconstructs a PointerTensor given it's attributes in form of a dictionary.
     We use the spread operator to pass the dict data as arguments
     to the init method of PointerTensor
-
     Args:
         Dict: a dictionary holding the attributes of the PointerTensor
```
[the remark this answers is not in the document]
Comment (iamtrask) [reply syft/serde.py]: Done!

File: syft/serde.py (hunk 1 above)
Comment (iamtrask) [reply syft/serde.py]: Changed!

File: syft/serde.py [hunk 6]
```diff
@@ -496,111 +499,60 @@ def _detail_ellipsis(ellipsis: bytes) -> Ellipsis:
...
-# PointerTensor
-
-
-def _simplify_pointer_tensor(ptr: PointerTensor) -> Dict:
+def _simplify_pointer_tensor(ptr: PointerTensor) -> tuple:
     """
     This function takes the attributes of a PointerTensor and saves them in a dictionary
-
     Args:
         PointerTensor: a PointerTensor
```
Comment (robert-wagner) [inline syft/serde.py]: Should be ptr (PointerTensor) as above. Or just ptr because Napoleon can handle type annotations

File: syft/serde.py [hunk 7]
```diff
@@ -191,7 +195,7 @@ def _detail_torch_tensor(tensor: bin) -> torch.Tensor:
         cls=tensor,
         torch_tensor=True,
         owner=None,
-        id=None,
+        id=id,
```
Comment (robert-wagner) [inline syft/serde.py]: Should we serialize owners or is that a later concept. (Also do we even need owner for pointer tensor)

Comment (robert-wagner) [reply syft/serde.py]: Also relatedly we should have tests for tensors with None id and non None id

File: syft/serde.py [hunk 8]
```diff
@@ -457,7 +461,7 @@ def _simplify_slice(my_slice: slice) -> Tuple[int, int, int]:
     Returns:
         list: a list holding the start, stop and step values
```
Comment (robert-wagner) [inline syft/serde.py]: It is not letting me review on the actual line but the arg for this function in the docs should be my_slice (slice). This is a problem throughout

File: syft/serde.py [hunk 9]
```diff
@@ -496,111 +497,60 @@ def _detail_ellipsis(ellipsis: bytes) -> Ellipsis:
...
+def _detail_pointer_tensor(tensor_tuple: tuple) -> PointerTensor:
     """
     This function reconstructs a PointerTensor given it's attributes in form of a dictionary.
     We use the spread operator to pass the dict data as arguments
     to the init method of PointerTensor
-
     Args:
-        Dict: a dictionary holding the attributes of the PointerTensor
-
+        tuple: a tuple holding the attributes of the PointerTensor
```
Comment (robert-wagner) [inline syft/serde.py]: should be tensor_tuple (tuple)

File: syft/workers/base.py [hunk 10]
```diff
@@ -193,15 +193,15 @@ def send(self, tensor, workers, ptr_id=None):
             ptr_id = int(10e10 * random.random())
 
         # Send the object
-        self.send_obj((ptr_id, tensor), worker)
+        self.send_obj(tensor, worker)
 
         pointer = tensor.create_pointer(
-            owner=self, location=worker, id_at_location=ptr_id, register=True
+            owner=self, location=worker, id_at_location=tensor.id, register=True, ptr_id=ptr_id
```
Comment (robert-wagner) [inline syft/workers/base.py]: split these up so that one arg is on each line (I think autopep8 does this)

File: syft/workers/base.py [hunk 11]
```diff
@@ -193,15 +193,15 @@ def send(self, tensor, workers, ptr_id=None):
...
         )
 
         return pointer
 
-    def set_obj(self, obj_data):
+    def set_obj(self, obj):
         """This adds an object to the registry of objects.
 
         :Parameters:
```
Comment (robert-wagner) [inline syft/workers/base.py]: should this section be in the new style or is that beyond the scope of this pr

File: syft/serde.py (hunk 7 above)
Comment (iamtrask) [reply syft/serde.py]: We shouldn't ever need to serialize owners because it's always going to be changed to whoever does the deserializing. There might be edge cases but by default I think it makes sense to ignore them

File: syft/workers/base.py (hunk 11 above)
Comment (iamtrask) [reply syft/workers/base.py]: beyond the scope I think

File: syft/workers/base.py (hunk 10 above)
Comment (iamtrask) [reply syft/workers/base.py]: Done!

File: syft/serde.py (hunk 9 above)
Comment (iamtrask) [reply syft/serde.py]: Fixed

File: syft/serde.py (hunk 8 above)
Comment (iamtrask) [reply syft/serde.py]: fixed

File: syft/workers/base.py (hunk 11 above)
Comment (iamtrask) [reply syft/workers/base.py]: i figured style stuff could be multiple PRs (collaborative) - i fixed a few doc style things and a few other things as well.

File: syft/serde.py [hunk 12]
```diff
@@ -456,10 +456,10 @@ def _simplify_slice(my_slice: slice) -> Tuple[int, int, int]:
     This function creates a list that represents a slice.
 
     Args:
-        slice: a python slice
+        my_slice (tuple): a python slice
```
Comment (robert-wagner) [inline syft/serde.py]: Should the type of this be tuple?

File: syft/serde.py [hunk 13]
```diff
@@ -474,7 +474,7 @@ def _detail_slice(my_slice: Tuple[int, int, int]) -> slice:
     This function extracts the start, stop and step from a list.
 
     Args:
-        list: a list defining the slice parameters [start, stop, step]
+        my_slice (tuple): a list defining the slice parameters [start, stop, step]
```
Comment (robert-wagner) [inline syft/serde.py]: A tuple defining the slice parameters?

File: syft/workers/base.py (hunk 11 above)
Comment (robert-wagner) [reply syft/workers/base.py]: Sounds good

File: syft/serde.py (hunk 12 above)
Comment (iamtrask) [reply syft/serde.py]: good catch

File: syft/serde.py (hunk 13 above)
Comment (iamtrask) [reply syft/serde.py]: this one is correct

Comment (robert-wagner) [reply syft/serde.py]: This was a comment on the text of the comment
